### PR TITLE
Entry Model Tweaks

### PR DESCRIPTION
**Pull request by bakerstu**
I found I needed to close a few corner case gaps and tweak a few things.

### CODE REVIEW

File: src/utils/EntryModel.cxxtest [hunk 1]
```diff
@@ -183,6 +183,70 @@ TEST(EntryModelTest, InitValuePopBack)
...
+    EXPECT_TRUE(uem.empty());
+
+    // unsigned 0 value with leading zeros, non-zero result
+    uem.init(4, 10, 0);
+
+    EXPECT_EQ(4U, uem.max_size());
+    uem.push_back_char('0');
+    EXPECT_EQ("0", uem.get_string());
+    EXPECT_EQ("   0", uem.get_string(true));
+    EXPECT_FALSE(uem.has_leading_zeros());
```
Comment (balazsracz) [inline src/utils/EntryModel.cxxtest:195]: consider adding expectations on uem.get_value() as well in these steps

File: src/utils/EntryModel.hxx [hunk 2]
```diff
@@ -205,6 +205,49 @@ public:
...
+
+    /// Set the radix base.
+    /// @param base new radix base to set.
+    /// @param convert convert the current value, as a string, to the new base.
+    void set_base(int base, bool convert)
+    {
+        if (base != base_ && convert)
+        {
+            string str;
+            switch (base_)
```
Comment (balazsracz) [inline src/utils/EntryModel.hxx:219]: I think you could write this a lot easier:
```
    string str = get_string();
    set_base(base);
    if (std::is_signed<T>::value)
    {
        value_ = strtoll(str.c_str(), nullptr, base);
    }
    else
    {
        value_ = strtoull(str.c_str(), nullptr, base);
    }
```

File: src/utils/EntryModel.hxx [hunk 3]
```diff
@@ -357,14 +396,19 @@ public:
...
     }
 
-    /// Clamp the value at the min or max.
+    /// Clamp the value at the min or max. Clamping will not occur if the value
+    /// is zero and there is space for more leading zeros.
     /// @param force Normally, clamping doesn't occur if the entry is "empty".
     ///              However, if force is set to true, we will clamp anyways.
     virtual void clamp(bool force = false)
     {
+        if (value_ == 0 && size_ < maxSize_)
```
Comment (balazsracz) [inline src/utils/EntryModel.hxx:405]: should this take into account "force" ?

**Review by balazsracz — APPROVED** [review]
